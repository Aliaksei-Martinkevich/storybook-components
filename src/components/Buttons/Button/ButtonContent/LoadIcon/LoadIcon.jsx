import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './LoadIcon.scss';

const LoadIcon = ({ iconUri, className }) => (
  <img
    className={classNames('load-icon', className)}
    alt=""
    src={iconUri}
  />
);

LoadIcon.defaultProps = {
  className: '',
};

LoadIcon.propTypes = {
  iconUri: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default LoadIcon;
