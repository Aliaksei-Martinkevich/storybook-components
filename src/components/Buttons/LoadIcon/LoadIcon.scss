@keyframes spin {
    to {
        transform: rotate(360deg);
    }
}

 
.load-icon {
    height: 18px;
    margin: auto;
    margin-right: 3%;
    animation: spin 1.5s steps(12, start) infinite;
}