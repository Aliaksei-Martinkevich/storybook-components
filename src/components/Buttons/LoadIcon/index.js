import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const LoadIcon = ({iconUri}) => (
<img 
  className={'load-icon'} 
  alt='' 
  src={iconUri}>
</img>
);

LoadIcon.propTypes = {
  iconUri: PropTypes.string.isRequired,
}

export default LoadIcon;
