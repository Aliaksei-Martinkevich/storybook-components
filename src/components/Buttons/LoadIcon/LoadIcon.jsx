import React from 'react';
import PropTypes from 'prop-types';
import './LoadIcon.scss';

const LoadIcon = ({ iconUri }) => (
  <img
    className="load-icon"
    alt=""
    src={iconUri}
  />
);

LoadIcon.propTypes = {
  iconUri: PropTypes.string.isRequired,
};

export default LoadIcon;
