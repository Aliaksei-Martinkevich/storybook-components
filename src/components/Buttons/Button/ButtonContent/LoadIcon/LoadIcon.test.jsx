import React from 'react';
import renderer from 'react-test-renderer';
import LoadIcon from './LoadIcon.jsx';
import icon from '../../../../../../assets/icons/svg/loading-circle-dashes-black.svg';

it('Should render without fails', () => {
  const customClassName = 'class-name-example';
  const component = renderer.create(<LoadIcon className={customClassName} iconUri={icon} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
