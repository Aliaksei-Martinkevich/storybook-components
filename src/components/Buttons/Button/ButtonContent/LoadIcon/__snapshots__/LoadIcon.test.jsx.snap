// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load icon Should render without fails 1`] = `
<img
  alt=""
  className="load-icon class-name-example"
  src="test-file-stub"
/>
`;
